docs/csharp/programming-guide/concepts/async/async-return-types.md

```diff
@@ -1,63 +1,74 @@
-  
-> [!IMPORTANT]
-> The <xref:System.Threading.Tasks.Task%601.Result%2A> property is a blocking property. If you try to access it before its task is finished, the thread that's currently active is blocked until the task completes and the value is available. In most cases, you should access the value by using `await` instead of accessing the property directly. <br/> The previous example retrieved the value of the <xref:System.Threading.Tasks.Task%601.Result%2A> property to block the main thread so that the `ShowTodaysInfo` method could finish execution before the application ended.  
+For more information about async methods, see [Asynchronous programming with async and await (C#)](./index.md).
+
+Several other types also exist that are specific to Windows workloads:
+
+- <xref:System.Windows.Threading.DispatcherOperation>, for async operations limited to Windows.
+- <xref:Windows.Foundation.IAsyncAction>, for async actions in UWP that do not return a value.
+- <xref:Windows.Foundation.IAsyncActionWithProgress%601>, for async actions in UWP that report progress but do not return a value.
+- <xref:Windows.Foundation.IAsyncOperation%601>, for async operations in UWP that return a value.
+- <xref:Windows.Foundation.IAsyncOperationWithProgress%602>, for async operations in UWP that report progress and return a value.
+
+## Task return type
+
+Async methods that don't contain a `return` statement or that contain a `return` statement that doesn't return an operand usually have a return type of <xref:System.Threading.Tasks.Task>. Such methods return `void` if they run synchronously. If you use a <xref:System.Threading.Tasks.Task> return type for an async method, a calling method can use an `await` operator to suspend the caller's completion until the called async method has finished.
+
+In the following example, the `WaitAndApologizeAsync` method doesn't contain a `return` statement, so the method returns a <xref:System.Threading.Tasks.Task> object. Returning a `Task` enables `WaitAndApologizeAsync` to be awaited. The <xref:System.Threading.Tasks.Task> type doesn't include a `Result` property because it has no return value.
+
+:::code language="csharp" source="async-return-types/async-returns2.cs" id="TaskReturn":::
+
+`WaitAndApologizeAsync` is awaited by using an await statement instead of an await expression, similar to the calling statement for a synchronous void-returning method. The application of an await operator in this case doesn't produce a value.
```

Comment:
It looks like this was confusing in the original article (https://stackoverflow.com/questions/12502917/await-statement-vs-expression). It may be worth a note to clarify that. Something like:

- By await statement, we mean that the result after awaiting is avoid. That means the awaited expression is `Task`.
- By await expression, we meant that the result after awaiting has a type. That means the awaited expression is `Task<T>`.